modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -14,6 +14,105 @@
 
 namespace opencv_test
 {
+// Draw random ellipses on given mat of given size and type
+static void initMatForFindingContours(cv::Mat& mat, const cv::Size& sz, const int type)
+{
+    cv::RNG& rng = theRNG();
+    mat = cv::Mat(sz, type, cv::Scalar::all(0));
+    size_t numEllipses = rng.uniform(1, 10);
+
+    for( size_t i = 0; i < numEllipses; i++ )
+    {
+        cv::Point center;
+        cv::Size  axes;
+        center.x     = rng.uniform(0, sz.width);
+        center.y     = rng.uniform(0, sz.height);
+        axes.width   = rng.uniform(2, sz.width);
+        axes.height  = rng.uniform(2, sz.height);
+        int color    = rng.uniform(1, 256);
+        double angle = rng.uniform(0., 180.);
+        cv::ellipse(mat, center, axes, angle, 0., 360., color, 1, FILLED);
+    }
+}
+
+enum WithHierarchy {NO_HIERARCHY, WITH_HIERARCHY};
+
+template<WithHierarchy withHierarchy = NO_HIERARCHY>
+cv::GComputation findContoursTestGAPI(const cv::Mat& in, const cv::RetrievalModes mode,
+                                      const cv::ContourApproximationModes method,
+                                      cv::GCompileArgs&& args,
+                                      std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+                                      std::vector<cv::Vec4i>&,
+                                      const cv::Point& offset = cv::Point())
+{
+    cv::GMat g_in;
+    cv::GOpaque<cv::Point> gOffset;
+    cv::GArray<cv::GArray<cv::Point>> outCts;
+    outCts = cv::gapi::findContours(g_in, mode, method, gOffset);
+    cv::GComputation c(GIn(g_in, gOffset), GOut(outCts));
+    c.apply(gin(in, offset), gout(out_cnts_gapi), std::move(args));
+    return c;
+}
+
+template<> cv::GComputation findContoursTestGAPI<WITH_HIERARCHY> (
```

Comment:
From my point of view need merge this functions if no future versions of findContours expected.

Reply:
No, these split is intentional: `findContours` has different amount of running code inside depending on whether we calculate hierarchy or not. And since the graph has different signatures for two cases, all the `GComputation` code should be also divided,  which leads to no profit in uniting the specializations

Reply:
Ok.

Reply:
Ok.